File: templates/java/java/spring_jpa_repo.java
#!target:java/{{.global.pkg}}/repo/{{.table.Title}}Repository.java
package {{pkg "java" .global.pkg}}.repo;

import {{pkg "java" .global.pkg}}.pojo.{{.table.Title}}{{.global.entity_suffix}}
import org.springframework.data.jpa.repository.JpaRepository
{{$pkType := type "java" .table.PkType}}
/** {{.table.Comment}}仓储接口 */
public interface {{.table.Title}}Repository : JpaRepository<{{.table.Title}}{{.global.entity_suffix}}, {{$pkType}}>{

}

File: templates/java/spring/spring_jpa_repo.kt
#!target:spring/src/main/kotlin/{{.global.pkg}}/repo/{{.table.Title}}JpaRepository.kt
package {{pkg "java" .global.pkg}}.repo;

import {{pkg "kotlin" .global.pkg}}.pojo.{{.table.Title}}Entity
import org.springframework.data.jpa.repository.JpaRepository
{{$pkType := type "kotlin" .table.PkType}}
/** {{.table.Comment}}仓储接口  */
interface {{.table.Title}}JpaRepository : JpaRepository<{{.table.Title}}Entity, {{$pkType}}> {

}